fhir-server/src/main/java/org/linuxforhealth/fhir/server/util/RestAuditLogger.java

```diff
@@ -496,71 +497,29 @@ private static void logBundleBatch(AuditLogService auditLogSvc, HttpServletReque
         Response.Status responseStatus) throws Exception {
-        AuditLogEntry entry = initLogEntry(AuditLogEventType.FHIR_BUNDLE);
-        long readCount = 0;
-        long createCount = 0;
-        long updateCount = 0;
-        long deleteCount = 0;
-        long executeCount = 0;
-        HTTPVerb requestMethod;
-
-        populateAuditLogEntry(entry, request, null, startTime, endTime, responseStatus);
-        if (requestBundle != null) {
+        
+        if (requestBundle == null) {
+            return;
+        }
+        AuditLogEntry entry = null;
+        // log individual audit event messages when the batch transaction is successful. 
+        if (responseStatus == Status.OK) {
             // We need the requestBundle so we know what the request was for at this point.
             // We don't have a "request" field otherwise
-            for (Entry bundleEntry : requestBundle.getEntry()) {
-                if (bundleEntry.getRequest() != null && bundleEntry.getRequest().getMethod() != null) {
-                    requestMethod = bundleEntry.getRequest().getMethod();
-                    boolean operation =  bundleEntry.getRequest().getUrl().getValue().contains("$")
-                                            || bundleEntry.getRequest().getUrl().getValue().contains("/%24");
-                    switch (HTTPVerb.Value.from(requestMethod.getValue())) {
-                    case GET:
-                        if (operation) {
-                            executeCount++;
-                        } else {
-                            readCount++;
-                        }
-                        break;
-                    case POST:
-                        if (operation) {
-                            executeCount++;
-                        } else {
-                            createCount++;
-                        }
-                        break;
-                    case PUT:
-                        updateCount++;
-                        break;
-                    case DELETE:
-                        if (operation) {
-                            executeCount++;
-                        } else {
-                            deleteCount++;
-                        }
-                        break;
-                    default:
-                        break;
-                    }
-                }
+            Iterator<Bundle.Entry> iter = requestBundle.getEntry().iterator();
+            for (Entry bundleEntry : responseBundle.getEntry()) {
+                Bundle.Entry requestEntry = iter.next();
+                entry = initLogEntry(AuditLogEventType.FHIR_BUNDLE);
+                populateAuditLogEntry(entry, request, bundleEntry.getResource(), startTime, endTime, responseStatus);
+                entry.setDescription("FHIR Bundle Transaction request");
+                entry.getContext().setAction(selectActionForBundleEntry(requestEntry));
+                auditLogSvc.logEntry(entry);
```

Comment:
why not just use bundleEntry here and remove the "iter" and "requestEntry" variables?

Reply:
requestEntry is being used to get the request method of Bundle entry in                "entry.getContext().setAction(selectActionForBundleEntry(requestEntry));"

Reply:
i think i missed the fact that requestEntry is from the requestBundle whereas the bundle entry is from the **responseBundle**